src/utils/html.js

```diff
@@ -403,14 +399,41 @@ u.convertToHyperlink = function (url) {
         return `<a target="_blank" rel="noopener" class="open-chatroom" href="${url}">${u.escapeHTML(pretty_url)}</a>`;
     }
     return `<a target="_blank" rel="noopener" href="${url}">${u.escapeHTML(pretty_url)}</a>`;
+};
+
+function isProtocolApproved (protocol, safeProtocolsList = APPROVED_URL_PROTOCOLS) {
+    return !!safeProtocolsList.includes(protocol);
+}
+
+// Will return false if URL is malformed or contains disallowed characters
+function isUrlValid (urlString) {
+    try {
+        const url = new URL(urlString);
+        return !!url;
+    } catch (error) {
+        return false;
+    }
 }
 
+u.convertUrlToHyperlink = function (url) {
+    const uri = getURI(url);
+    if (uri !== null && isUrlValid(url) && isProtocolApproved(uri._parts.protocol)) {
```

Comment:
The `uri !== null` check can go into `isUrlValid`, after all, `null` isn't a valid URL :)

Generally in the Converse codebase we write abbreviations as all-caps. So `isURLValid` instead of `isUrlValid`.

Reply:
`uri` is not a url string, but an instance of the `URI` class, whereas the purpose of `isUrlValid` is to check for valid url strings. I moved this check to here from the code that you commented on below, which was then removed.

Reply:
Ah sorry, I see now they're two different variables.